File: 102-binary-tree-level-order-traversal/binary-tree-level-order-traversal.rb
# Definition for a binary tree node.
# class TreeNode
#     attr_accessor :val, :left, :right
#     def initialize(val = 0, left = nil, right = nil)
#         @val = val
#         @left = left
#         @right = right
#     end
# end
# @param {TreeNode} root
# @return {Integer[][]}
def level_order(root)
    return [] unless root
    queue = []
    level = []
    queue << root
    while queue.size > 0 do
        len = queue.size
        temp = []
        len.times do
            el = queue.shift
            temp << el.val
            queue << el.left if el.left
            queue << el.right if el.right
        end
        level << temp
    end
    level
end

File: 102-binary-tree-level-order-traversal/binary-tree-level-order-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        queue = deque()
        if root:
            queue.append(root)
        ans = []
        while queue:
            small_ans = []
            length = len(queue)
            while length > 0:
                node = queue.popleft()
                if node.left:
                    queue.append(node.left)
                if node.right:
                    queue.append(node.right)
                small_ans.append(node.val)
                length -= 1
            ans.append(small_ans)
        return ans